packages/react-components/src/components/search-bar/search-button.tsx

```diff
@@ -0,0 +1,135 @@
+import { keyframes } from '@emotion/core'
+import styled from '@emotion/styled'
+import React from 'react'
+import useThrottle from 'react-use/lib/useThrottle'
+import SearchIcon_ from './search-icon'
+import { getSize, SearchTheme } from './theme'
+
+const time = '2s'
+const purp = '#9933FF'
+const pink = '#FF6666'
+const strongEasing = 'cubic-bezier(0.920, 0.240, 0.185, 0.730)'
+
+const searchFx = keyframes`
+    0% {
+      transform: rotate(34deg) translate(-10px, 80px);
+    };
+    
+    100% {
+      transform: rotate(34deg) translate(-10px, -20px);
+    }
+`
+const plus = keyframes`
+    0% { 
+      transform: translate(0px, 0px);
+      opacity: 0;
+    }
+    50% {
+      opacity: 1;
+    }
+    100% {
+      transform: translate(10px, -17px);
+      opacity: 0;
+    }
+`
+const gradientFade = keyframes`
+    0% {
+      opacity: 0;
+      background-position: 0% 100%;
```

Comment:
@giannif is there any way to do this with a separate element that you can animate `transform` on? Not a huge deal, but animating `background-position` needs to repaint and could animate choppy.

Reply:
It looks this animation is on a pseudo element. Could you make it have `width: 400%` and then animate it's `transform` instead?

Reply:
@sketchybones I'll give it a shot! 👍

Reply:
@sketchybones look better?